### PR TITLE
[Feature][E2E] Add mysql-cdc e2e testcase

### PR DESCRIPTION
**Pull request by zhilinli123**
issues:https://github.com/apache/incubator-seatunnel/issues/4603

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        MySqlContainer mySqlContainer =
+                new MySqlContainer(version)
+                        .withConfigurationOverride("docker/server-gtids/my.cnf")
+                        .withSetupSQL("docker/setup.sql")
+                        .withNetwork(NETWORK)
+                        .withNetworkAliases(MYSQL_HOST)
+                        .withDatabaseName(MYSQL_DATABASE)
+                        .withUsername(MYSQL_USER_NAME)
+                        .withPassword(MYSQL_USER_PASSWORD)
+                        .withLogConsumer(new Slf4jLogConsumer(LOG));
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]:
```suggestion
                        .withLogConsumer(new Slf4jLogConsumer( DockerLoggerFactory.getLogger("mysql-docker-image)));
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SPARK, EngineType.FLINK},
+        disabledReason = "")
+public class MysqlCDCIT extends TestSuiteBase implements TestResource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MysqlCDCIT.class);
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                        .withLogConsumer(new Slf4jLogConsumer(LOG));
+        return mySqlContainer;
+    }
+
+    @BeforeAll
+    @Override
+    public void startUp() throws ClassNotFoundException, InterruptedException {
+        LOG.info("The second stage: Starting Mysql containers...");
+        Startables.deepStart(Stream.of(MYSQL_CONTAINER)).join();
+        LOG.info("Mysql Containers are started");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]:
```suggestion
        log.info("Mysql Containers are started");
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.awaitility.Awaitility.await;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SPARK, EngineType.FLINK},
+        disabledReason = "")
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]:
```suggestion
        disabledReason = "Currently SPARK and FLINK do not support cdc")
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        LOG.info("Mysql Containers are started");
+        inventoryDatabase.createAndInitialize();
+        LOG.info("Mysql ddl execution is complete");
+    }
+
+    @TestTemplate
+    public void test(TestContainer container) throws IOException, InterruptedException {
+        LOG.info("-------docker mysql host:{}", MYSQL_CONTAINER.getHost());
+        LOG.info("-------docker mysql port:{}", MYSQL_CONTAINER.getDatabasePort());
+        LOG.info("-------docker mysql database:{}", MYSQL_CONTAINER.getDatabaseName());
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: remove？

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+        CompletableFuture<Void> executeJobFuture =
+                CompletableFuture.supplyAsync(
+                        () -> {
+                            try {
+                                container.executeJob("/mysqlcdc_to_console.conf");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: rename `mysqlcdc_to_console.conf` to `mysqlcdc_to_jdbc.conf`
```suggestion
                                container.executeJob("/mysqlcdc_to_jdbc.conf");
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                            } catch (Exception e) {
+                                LOG.error("Commit task exception :" + e.getMessage());
+                                throw new RuntimeException(e);
+                            }
+                            return null;
+                        });
+        await().atMost(60000, TimeUnit.MILLISECONDS)
+                .untilAsserted(
+                        () -> {
+                            log.info(querySql(SINK_SQL).toString());
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java:133]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+
+        // insert update delete
+        upsertDeleteSourceTable();
+
+        // stream stage
+        await().atMost(60000, TimeUnit.MILLISECONDS)
+                .untilAsserted(
+                        () -> {
+                            Assertions.assertEquals(10, querySql(SINK_SQL).size());
+                        });
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: duplicate check
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+# #   MySQL-CDC {
+#       result_table_name = "fake"
+#       parallelism = 1
+#       server-id = 5656
+#       username = "mysqluser"
+#       password = "mysqlpw"
+#       table-names = ["inventory_vwyw0n.products"]
+#       base-url = "jdbc:mysql://localhost:56725/inventory_vwyw0n"
+#     }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    driver = "com.mysql.cj.jdbc.Driver"
+    user = "st_user"
+    password = "seatunnel"
+
+    generate_sink_sql = true
+    # You need to configure both database and table
+    database = mysql_cdc
+    table = products_cdc_sink
+    primary_keys = ["id"]
+}
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf]:
```suggestion
  jdbc {
    source_table_name = "customers_mysql_cdc"
    url = "jdbc:mysql://mysql_cdc_e2e:3306/mysql_cdc"
    driver = "com.mysql.cj.jdbc.Driver"
    user = "st_user"
    password = "seatunnel"

    generate_sink_sql = true
    # You need to configure both database and table
    database = mysql_cdc
    table = products_cdc_sink
    primary_keys = ["id"]
  }
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+#       username = "mysqluser"
+#       password = "mysqlpw"
+#       table-names = ["inventory_vwyw0n.products"]
+#       base-url = "jdbc:mysql://localhost:56725/inventory_vwyw0n"
+#     }
+
+source {
+  MySQL-CDC {
+    result_table_name = "customers_mysql_cdc"
+    parallelism = 1
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/mysqlcdc_to_console.conf]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.cdc.mysql;
+
+import org.apache.seatunnel.connectors.seatunnel.cdc.mysql.testutils.MySqlContainer;
+import org.apache.seatunnel.connectors.seatunnel.cdc.mysql.testutils.MySqlVersion;
+import org.apache.seatunnel.connectors.seatunnel.cdc.mysql.testutils.UniqueDatabase;
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java:22]: move to `connector-cdc-mysql-e2e` from `connector-cdc-mysql`

**Comment by hailin0** [comment]
check ci error

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java (hunk 12 above)
Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java:22]: Done

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/pom.xml [hunk 13]
```diff
@@ -82,24 +82,4 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-
-    <build>
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/pom.xml]: Or, fix find connector jar package logic to ignore test.jar

org.apache.seatunnel.e2e.common.util.ContainerUtil#copyConnectorJarToContainer

**Comment by hailin0** [comment]
merge dev branch

**Comment by zhilinli123** [comment]
> merge dev branch
Have merged help run CI
@hailin0

**Comment by zhilinli123** [comment]
Help re-run CI

**Review by hailin0 — DISMISSED** [review]

**Comment by hailin0** [comment]
LGTM

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/ddl/mysql_cdc.sql [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+--
+
+-- ----------------------------------------------------------------------------------------------------------------
+-- DATABASE:  inventory
+-- ----------------------------------------------------------------------------------------------------------------
+CREATE DATABASE IF NOT EXISTS `mysql_cdc`;
+
+use mysql_cdc;
+-- Create a mysql data source table
+CREATE TABLE mysql_cdc.products_cdc (
```
Comment (EricJoy2048) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/resources/ddl/mysql_cdc.sql]: create table if not exists test1000004
(
    id                   int auto_increment primary key,
    f_binary             binary(64)          null,
    f_blob               blob                null,
    f_long_varbinary     mediumblob          null,
    f_longblob           longblob            null,
    f_tinyblob           tinyblob            null,
    f_varbinary          varbinary(100)      null,
    f_smallint           smallint            null,
    f_smallint_unsigned  smallint unsigned   null,
    f_mediumint          mediumint           null,
    f_mediumint_unsigned mediumint unsigned  null,
    f_int                int                 null,
    f_int_unsigned       int unsigned        null,
    f_integer            int                 null,
    f_integer_unsigned   int unsigned        null,
    f_bigint             bigint              null,
    f_bigint_unsigned    bigint unsigned     null,
    f_numeric            decimal             null,
    f_decimal            decimal             null,
    f_float              float               null,
    f_double             double              null,
    f_double_precision   double              null,
    f_longtext           longtext            null,
    f_mediumtext         mediumtext          null,
    f_text               text                null,
    f_tinytext           tinytext            null,
    f_varchar            varchar(100)        null,
    f_date               date                null,
    f_datetime           datetime            null,
    f_timestamp          timestamp           null,
    f_bit1 BIT(1) NULL,
    f_bit64 BIT(64) NULL,
    f_char CHAR NULL,
    f_enum ENUM('enum1', 'enum2', 'enum3') NULL,
    f_mediumblob MEDIUMBLOB NULL,
    f_long_varchar LONG VARCHAR NULL,
    f_real REAL NULL,
    f_time TIME NULL,
    f_tinyint TINYINT NULL,
    f_tinyint_unsigned TINYINT UNSIGNED NULL,
    f_json json NULL，
    f_year  year null,
    f_set set('set1', 'set2', 'set3') null
);

**Comment by ic4y** [comment]
I suggest covering the data types fixed in these two PRs:
https://github.com/apache/incubator-seatunnel/pull/4670
TIME
https://github.com/apache/incubator-seatunnel/pull/4671
tinyint(1) bit(1) bit(8) bit(32) bit(64)

**Comment by zhilinli123** [comment]
> I suggest covering the data types fixed in these two PRs: #4670 TIME #4671 tinyint(1) bit(1) bit(8) bit(32) bit(64)

Can you refer to  : https://ververica.github.io/flink-cdc-connectors/master/content/connectors/mysql-cdc.html
Mysql Cdc at present Many types are currently not supported

**Comment by ic4y** [comment]
Wait for https://github.com/apache/incubator-seatunnel/pull/4671 to be merged and JDBC MySQL YEAR type support, then add corresponding data type tests

**Comment by EricJoy2048** [comment]
> Wait for #4671 to be merged and JDBC MySQL YEAR type support, then add corresponding data type tests

#4671  is merged.

**Comment by zhilinli123** [comment]
> > 等待#4671合并和JDBC MySQL YEAR类型支持，然后添加相应的数据类型测试
> 
> #4671 合并。



> > Wait for #4671 to be merged and JDBC MySQL YEAR type support, then add corresponding data type tests
> 
> #4671 is merged.

I'm going to support year first and refine the type
https://github.com/apache/incubator-seatunnel/issues/4695

**Comment by zhilinli123** [comment]
> > Wait for #4671 to be merged and JDBC MySQL YEAR type support, then add corresponding data type tests
> 
> #4671 is merged.

I'm going to support year first and refine the type
https://github.com/apache/incubator-seatunnel/issues/4695

**Comment by zhilinli123** [comment]
@ic4y @hailin0 @EricJoy2048  Please take a look thanks

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+                    + " f_json, f_year from mysql_cdc_e2e_source_table";
+    private static final String SINK_SQL =
+            "select id, cast(f_binary as char) as f_binary, cast(f_blob as char) as f_blob, cast(f_long_varbinary as char) as f_long_varbinary,"
+                    + " cast(f_longblob as char) as f_longblob, cast(f_tinyblob as char) as f_tinyblob, cast(f_varbinary as char) as f_varbinary,"
+                    + " f_smallint, f_smallint_unsigned, f_mediumint, f_mediumint_unsigned, f_int, f_int_unsigned, f_integer, f_integer_unsigned,"
+                    + " f_bigint, f_bigint_unsigned, f_numeric, f_decimal, f_float, f_double, f_double_precision, f_longtext, f_mediumtext,"
+                    + " f_text, f_tinytext, f_varchar, f_date, f_datetime, f_timestamp, f_bit1, cast(f_bit64 as char) as f_bit64, f_char,"
+                    + " f_enum, cast(f_mediumblob as char) as f_mediumblob, f_long_varchar, f_real, f_time, f_tinyint, f_tinyint_unsigned,"
+                    + " f_json, cast(f_year as year) from mysql_cdc_e2e_source_table";
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: sink table is `mysql_cdc_e2e_source_table`？

Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: > sink table is `mysql_cdc_e2e_source_table`？

Sorry for my negligence.  I have corrected it.  Thank you for your review

Comment (zhilinli123) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java]: @hailin0 Thank you ci review I have changed it. Thank you for your careful review

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**